pdns/lua-record.cc

```diff
@@ -1095,19 +1096,14 @@ static void setupLuaRecords(LuaContext& lua) // NOLINT(readability-function-cogn
    *
    * @example ifportup(443, { '1.2.3.4', '5.4.3.2' })"
    */
-  lua.writeFunction("ifportup", [](int port, const boost::variant<iplist_t, ipunitlist_t>& ips, const boost::optional<std::unordered_map<string,string>> options) {
-      if (port < 0) {
-        port = 0;
-      }
-      if (port > std::numeric_limits<uint16_t>::max()) {
-        port = std::numeric_limits<uint16_t>::max();
-      }
+  lua.writeFunction("ifportup", [](int port, const boost::variant<iplist_t, ipunitlist_t>& ips, const boost::optional<std::unordered_map<string, string>>& options) {
```

Comment:
Why did you change the signature of the lambda here? (passing `options` by reference)

This is not related to the algorithmic changes.

Reply:
Looked like a mistake

Reply:
I'm afraid it actually breaks the feature because the `options` parameter is no longer optional.